lib/Graph/Nodes.cpp

```diff
@@ -265,8 +265,6 @@ static void verifyRelu(NodeValue result, NodeValue input) {
   if (input.getType()->isQuantizedType()) {
     assert(result.getType()->isQuantizedType());
     checkSameShape(result, input);
-    assert(result.getType()->getOffset() == -128 &&
```

Comment:
There are currently 3 quantization schemas: asymmetric, symmetric, and symmetric with uint8.
If you use regular symmetric schema (not the one with uint8) you'll end up wasting 1 bit for relu and hence acquire higher accuracy loss.
I'm not really sure if it's a good idea. Do you have a use case that would want this and not the one with uint8 or asymmetric?

Reply:
Yeah it's not a good idea just for RELU by itself. But we currently only allow a single quantization schema to be set for an entire network. So for example for Resnet50, we cannot use symmetric quantization because of this assert -- if you profile with symmetric and then load the profile to quantize with it then this assertion fails.

Additionally, different backends might only support symmetric quantization. Or it could be faster/more efficient to execute with symmetric quantization since it's simpler to implement than asymmetric. So I think we should allow for this even if there are obvious downsides, as we don't know what the backend supports or doesn't.

Reply:
@rdzhabarov I think that as soon as the `symmetric` schema is requested by the user, the offset should by 0 for all quantized tensors, even if it would  result in a lower accuracy. It's a user choice, potentially dictated by the targeted hardware. If Glow fails compiling with this `symmetric` schema as soon as a RELU is present in the model, I think it's an issue. `symmetric with uint8` is available for user who want to gain 1 bit at the output of RELU.

Reply:
@jfix71 @tlepley makes sense.